engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/plan/NadelExecutionPlanFactory.kt

```diff
@@ -66,11 +67,15 @@ internal class NadelExecutionPlanFactory(
     companion object {
         fun create(
             executionBlueprint: NadelOverallExecutionBlueprint,
+            javaCompatTransforms: List<Any>,
             engine: NextgenEngine,
         ): NadelExecutionPlanFactory {
+            val customNadelTransforms = (javaCompatTransforms as List<NadelTransformJavaCompat<Any>>)
```

Comment:
We shouldn't assume that anything coming into the nextgen engine is a Java compat. What if one day we have Kotlin in the AGG codebase? Then if we feed in a pure `NadelTransform` instance here, this would crash. This is exactly why we should never use `Object` and unsafe casts if it's possible to use something more specific.

We need to have _type safety_ here. We should take in `List<NadelTransform>`. If anyone needs to feed in a `JavaCompat` that's their issue. In the engine we shouldn't deal with it.

In the AGG code we should have:

```
Nadel.newNadel()
  .engineFactory((nadel) -> {
    List<NadelTransform> transforms = FpKit.map(
      List.of(…),
      NadelTransformJavaCompat::create,
    );
    return new NextgenEngine(
      nadel,
      transforms,
    );
  })
  .build();
```